quickwit/quickwit-cli/src/logger.rs

```diff
@@ -116,3 +119,75 @@ pub fn setup_logging_and_tracing(
         Ok(())
     }))
 }
+
+enum EventFormat<'a> {
+    Full(Format<Full, UtcTime<Vec<BorrowedFormatItem<'a>>>>),
+    Json(tracing_subscriber::fmt::format::Format<tracing_subscriber::fmt::format::Json>),
+}
```

Comment:
Is there any strong reason for using a custom event format rather than a [boxed layer](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/layer/trait.Layer.html#method.boxed)? Maybe from a performance point of view enum is a tiny bit better, but boxing would be substantially simpler.
https://github.com/quickwit-oss/quickwit/blob/f9466262c559220cbef0c172cef07bc79cad731d/quickwit/quickwit-lambda/src/logger.rs#L92-L102

Reply:
I assumed we didn't go for a boxed implementation in the first place for a good reason, so I also added unboxed code. I'm open to a refactor that uses the same code in `cli` and `lambda` and potentially uses boxed layers.